server/sonar-server/src/main/java/org/sonar/server/computation/step/IntegrateIssuesStep.java

```diff
@@ -0,0 +1,151 @@
+  private void processIssues(Component component) {
+    Tracking<DefaultIssue, DefaultIssue> tracking = tracker.track(component);
+    DiskCache<DefaultIssue>.DiskAppender cacheAppender = issueCache.newAppender();
+    try {
+      issueVisitors.beforeComponent(component, tracking);
+      fillNewOpenIssues(component, tracking, cacheAppender);
+      fillExistingOpenIssues(component, tracking, cacheAppender);
+      closeUnmatchedBaseIssues(component, tracking, cacheAppender);
+      issueVisitors.afterComponent(component);
+    } finally {
+      cacheAppender.close();
+    }
+  }
+
+  private void fillNewOpenIssues(Component component, Tracking<DefaultIssue, DefaultIssue> tracking, DiskCache<DefaultIssue>.DiskAppender cacheAppender) {
+    Set<DefaultIssue> issues = tracking.getUnmatchedRaws();
+    for (DefaultIssue issue : issues) {
+      issueLifecycle.initNewOpenIssue(issue);
+      process(component, issue, cacheAppender);
+    }
+  }
+
+  private void fillExistingOpenIssues(Component component, Tracking<DefaultIssue, DefaultIssue> tracking, DiskCache<DefaultIssue>.DiskAppender cacheAppender) {
+    for (Map.Entry<DefaultIssue, DefaultIssue> entry : tracking.getMatchedRaws().entrySet()) {
+      DefaultIssue raw = entry.getKey();
+      DefaultIssue base = entry.getValue();
+      issueLifecycle.mergeExistingOpenIssue(raw, base);
+      process(component, raw, cacheAppender);
+    }
+    for (Map.Entry<Integer, DefaultIssue> entry : tracking.getOpenManualIssuesByLine().entries()) {
+      int line = entry.getKey();
+      DefaultIssue manualIssue = entry.getValue();
+      manualIssue.setLine(line == 0 ? null : line);
+      process(component, manualIssue, cacheAppender);
+    }
+  }
+
+  private void closeUnmatchedBaseIssues(Component component, Tracking<DefaultIssue, DefaultIssue> tracking, DiskCache<DefaultIssue>.DiskAppender cacheAppender) {
+    for (DefaultIssue issue : tracking.getUnmatchedBases()) {
+      // TODO should replace flag "beingClosed" by express call to transition "automaticClose"
+      issue.setBeingClosed(true);
+      // TODO manual issues -> was updater.setResolution(newIssue, Issue.RESOLUTION_REMOVED, changeContext);. Is it a problem ?
+      process(component, issue, cacheAppender);
+    }
+  }
+
+  private void process(Component component, DefaultIssue issue, DiskCache<DefaultIssue>.DiskAppender cacheAppender) {
+    issueLifecycle.doAutomaticTransition(issue);
+    issueVisitors.onIssue(component, issue);
+    cacheAppender.append(issue);
+  }
+
+  private void closeIssuesForDeletedComponentUuids(Set<String> deletedComponentUuids) {
+    DiskCache<DefaultIssue>.DiskAppender cacheAppender = issueCache.newAppender();
+    try {
+      for (String deletedComponentUuid : deletedComponentUuids) {
+        List<DefaultIssue> issues = baseIssuesLoader.loadForComponentUuid(deletedComponentUuid);
+        for (DefaultIssue issue : issues) {
+          issue.setBeingClosed(true);
+          // TODO should be renamed
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)